### PR TITLE
feat: add support dataset.max_time_travel_hours

### LINKED ISSUE #1637: Add support for setting `max_time_travel_hours` when creating a dataset

### PR DESCRIPTION
**Pull request by Gaurang033**
Fixes #1637

### CODE REVIEW

File: google/cloud/bigquery/dataset.py [hunk 1]
```diff
@@ -527,13 +527,33 @@ class Dataset(object):
...
 
     def __init__(self, dataset_ref) -> None:
         if isinstance(dataset_ref, str):
             dataset_ref = DatasetReference.from_string(dataset_ref)
         self._properties = {"datasetReference": dataset_ref.to_api_repr(), "labels": {}}
 
+    @property
+    def max_time_travel_hours(self):
+        """
+        Optional[int]: Defines the time travel window in hours. The value can be from 48 to 168 hours (2 to 7 days).
```
Comment (Linchin) [inline google/cloud/bigquery/dataset.py:541]: Could you please also add in the comments that value must be an integer expressed in multiples of 24 (48, 72, 96, 120, 144, 168), as [indicated in the documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language)? Thank you!

File: google/cloud/bigquery/dataset.py [hunk 2]
```diff
@@ -527,13 +527,33 @@ class Dataset(object):
...
+        The default value is 168 hours if this is not set.
+        """
+        return self._properties.get("maxTimeTravelHours")
+
+    @max_time_travel_hours.setter
+    def max_time_travel_hours(self, hours):
```
Comment (Linchin) [inline google/cloud/bigquery/dataset.py:547]: Similarly, raise an error if the value is not multiples of 24.

File: google/cloud/bigquery/dataset.py (hunk 1 above)
Comment (Gaurang033) [reply google/cloud/bigquery/dataset.py:541]: @Linchin  done.

File: google/cloud/bigquery/dataset.py (hunk 2 above)
Comment (Gaurang033) [reply google/cloud/bigquery/dataset.py:548]: @Linchin  done

**Comment by Gaurang033** [comment]
@Linchin  added units tests.

File: google/cloud/bigquery/dataset.py [hunk 3]
```diff
@@ -501,7 +501,9 @@ def from_api_repr(cls, resource: dict) -> "AccessEntry":
         if len(entry) != 0:
             raise ValueError("Entry has unexpected keys remaining.", entry)
 
-        return cls(role, entity_type, entity_id)
+        config = cls(role, entity_type, entity_id)
+        config._properties = copy.deepcopy(resource)
+        return config
```
Comment (Linchin) [inline google/cloud/bigquery/dataset.py:506]: Could you please explain why this change is necessary?

Comment (Gaurang033) [reply google/cloud/bigquery/dataset.py]: @Linchin  this function should not be changed. Thanks for catching it. I reverted it back.

File: tests/system/test_client.py [hunk 4]
```diff
@@ -238,6 +238,11 @@ def test_create_dataset(self):
         self.assertEqual(dataset.dataset_id, DATASET_ID)
         self.assertEqual(dataset.project, Config.CLIENT.project)
 
+    def test_create_dataset_max_time_travel_hours(self):
```
[the remark this answers is not in the document]
Comment (Gaurang033) [reply tests/system/test_client.py:241]: I added following tests. 

- test_create_dataset_with_max_time_travel_hours
- test_create_dataset_with_max_time_travel_hours_not_multiple_of_24
- test_create_dataset_with_max_time_travel_hours_is_less_than_2_days
- test_create_dataset_with_max_time_travel_hours_is_greater_than_7_days
- test_create_dataset_with_max_time_travel_hours_not_is_not_int

File: google/cloud/bigquery/dataset.py [hunk 5]
```diff
@@ -527,13 +527,33 @@ class Dataset(object):
...
+        """
+        Optional[int]: Defines the time travel window in hours. The value can be from 48 to 168 hours (2 to 7 days).
+        The default value is 168 hours if this is not set.
+        """
+        return self._properties.get("maxTimeTravelHours")
+
+    @max_time_travel_hours.setter
+    def max_time_travel_hours(self, hours):
+        if not isinstance(hours, int):
+            raise ValueError("Pass the int value")
```
[the remark this answers is not in the document]
Comment (Gaurang033) [reply google/cloud/bigquery/dataset.py]: @Linchin  done and added a unit tests for that as well.

File: tests/unit/test_create_dataset.py [hunk 6]
```diff
@@ -466,3 +466,82 @@ def test_create_dataset_with_default_rounding_mode_if_value_is_in_possible_value
...
+    conn = client._connection = make_connection(resource)
+
+    ds_ref = DatasetReference(PROJECT, DS_ID)
+    before = Dataset(ds_ref)
+    before.max_time_travel_hours = 24 * 3
+    after = client.create_dataset(before)
+
+    assert after.dataset_id == DS_ID
+    assert after.project == PROJECT
+    assert after.default_rounding_mode is None
```
Comment (Linchin) [inline tests/unit/test_create_dataset.py:489]:
```suggestion
    assert after.max_time_travel_hours == 24 * 3
```

File: google/cloud/bigquery/dataset.py [hunk 7]
```diff
@@ -533,6 +534,27 @@ def __init__(self, dataset_ref) -> None:
             dataset_ref = DatasetReference.from_string(dataset_ref)
         self._properties = {"datasetReference": dataset_ref.to_api_repr(), "labels": {}}
 
+    @property
+    def max_time_travel_hours(self):
+        """
+        Optional[int]: Defines the time travel window in hours. The value can be from 48 to 168 hours (2 to 7 days).
+        value must in multiple of 24 hours (48, 72, 96, 120, 144, 168)
+        The default value is 168 hours if this is not set.
+        """
```
Comment (Linchin) [inline google/cloud/bigquery/dataset.py:543]:
```suggestion
    @property
    def max_time_travel_hours(self):
        """
        Optional[int]: Defines the time travel window in hours. The value can
        be from 48 to 168 hours (2 to 7 days), and in multiple of 24 hours
        (48, 72, 96, 120, 144, 168).
        The default value is 168 hours if this is not set.
        """
```

**Review by Linchin — CHANGES_REQUESTED** [review]

**Comment by Linchin** [comment]
Thank you @Gaurang033!

**Review by Linchin — APPROVED** [review]
